### DISCUSSION TITLE
Pro Tip: Fully uninstall and reinstall MA add-on to resolve YT music issues!

### DISCUSSION THREAD

**Discussion by TheWebMachine** [music-assistant/support#2055] [Show and tell]
**TL;DR: If YTM refuses to play some tracks, randomly halts the Queue or keeps repeating same track without moving on, or shows messages about invalid Track IDs in MA's add-on log, _try fully removing MA add-on, reinstalling, and setting back up again_. Simply removing/reconfiguring YTM provider is not enough, it would seem.**

Sometimes you gotta make a clean break for it...

I've been having random issues with YTM playback such as tracks randomly being unable to play with logs about the track not being available on YTM in the last few betas. (I failed to capture the actual log message in my haste, but it was saying something to the tune of "track ID blahblahblah couldn't be found on YTmusic-idblah")

Not even removing and re-adding YTM provider was clearing them up. Playback would just halt when it got to a track it couldn't find (despite being added as part of a valid YTM playlist) or it would just keep endlessly repeating the previous track before the missing one. I'd be working and then realize I've been hearing the same song on repeat for 30 minutes.

As I didn't want to believe that MA had gotten suddenly worse instead of better, and the fact that I saw no issues posted by other YTM users about this, I finally went with the nuclear option: Full removal of the Add-on container.

Upon reinstalling the add-on, setting everything back up, and letting YTM sync, everything has been fine with YTM playback so far.

I think something is being cached in MA for YTM that does *not* get deleted when YTM provider is removed, so it doesn't resolve issues with invalid track IDs in the MA database to simply remove YTM and set back up. However, fully removing the add-on in HA removes all add-on data, which seems to have cleared up my issues this time around.

For context, I've been on the beta journey since b76, so I may have just accumulated junk in MA through all those betas that finally clogged up the works. At any rate, it seems that at least some errors require a full MA reinstall to correct. I didn't have to touch the HA integration and only the MA add-on, itself, needed set back up.

If the issue presents itself again in the future, I'll be sure to capture the logs and submit a proper Issue. This post is just putting my experience out there in case someone else finds themselves beating their head off the wall about a similar issue they are having.
